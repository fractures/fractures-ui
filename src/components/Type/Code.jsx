import PropTypes from "prop-types"
import styled from "styled-components"

const propTypes = {
	children: PropTypes.any,
	className: PropTypes.string,
	isSmall: PropTypes.bool
}

const defaultProps = {
	isSmall: false
}

const Code = styled.code`
	color: var(--color-gray-700);

	font-size: ${ props => (props.isSmall ? "0.75rem" : "1rem;") };
	font-family: var(--font-mono);
	line-height: 1rem;
`

Code.propTypes = propTypes
Code.defaultProps = defaultProps

export default Code
