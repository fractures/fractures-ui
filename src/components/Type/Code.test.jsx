import Code from "./Code"

describe("Code", () => {
	const tree = renderer.create(<Code>Debug.Call()</Code>).toJSON()

	it("Matches snapshot", () => expect(tree).toMatchSnapshot())
})
