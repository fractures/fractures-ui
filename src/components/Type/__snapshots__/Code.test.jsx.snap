// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code Matches snapshot 1`] = `
<code
  className="sc-bdVaJa DzoOc"
>
  Debug.Call()
</code>
`;
